doc/design/cli.md

```diff
@@ -0,0 +1,105 @@
+
+```bash
+$ sqlflow
+sqlflow> _
+```
+
+To achieve this goal, we also want to hide implementation details about Docker because users don't have to know this. As a result, there are two possible solutions:
+1. Wrap the current SQLFlow REPL in a script, we already have such an [attempt](https://github.com/sql-machine-learning/sqlflow/pull/2114).
+1. Decouple the REPL implementation from the core code of SQLFlow and provide a native binary that only functions as a pure client.
+
+We decide to choose the second solution because it's more lightweight, if a user already has a SQLFlow deployment in her company, all she has to do is to download the native binary.
+
+## Design
+
+### Naming
+
+`REPL` is not a good name, We'll rename the binary as simply `sqlflow`.
+
+### Dependencies and Installation
+
+The original `repl`'s only dependency is the `it2check` bash script that determines whether to call `sixel` to render images.
+
+We'll provide installation scripts on Mac (Bash) and Windows (PowerShell) to install `sqlflow` and its dependencies swiftly. The four scripts are:
+
+1. install.sh (Install Docker, `sqlflow`, and the image `sqlflow/sqlflow:latest` on a Mac)
+1. install-client.sh (Only `sqlflow` on a Mac)
+1. install.ps1(Install Docker, `sqlflow`, and the image `sqlflow/sqlflow:latest` on a PC, coming later)
+1. install-client.ps1 (Only `sqlflow` on a PC, coming later)
+
+### Refactoring
+
+Fortunately, there'are only a few places in the REPL codebase that requires SQLFlow core.
+
+#### Workflow
+
+At the moment, SQLFlow workflow calls `repl` to execute a single step, after the refactoring, as `repl` has become a pure client, the workflow code should call the [`step` binary](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/step/step.go#L45-L53) instead.
+
+Consequently, we can delete code in `repl` about workflow safely:
+
+1. [`isWorkflowStep`](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L314-L320)
+
+1. [Table in protobuf](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L161-L163)
+
+1. [The step package](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L35)
+
+#### [Command Line Flags](https://github.com/sql-machine-learning/sqlflow/blob/be7c5728f47e8d3b81893c1d712974a6ddcd5f1c/cmd/repl/repl.go#L261-L268)
+
+1. We don't need the flag `--model_dir` any more because the training process is running in `sqlflowserver`.
+
+1. We'll define a new flag `--host` that holds the `sqlflowserver` address.
```

Comment:
Not `--host`, but `--sqlflow_server`.  Let us make sure all names are accurate and precise.

Reply:
Done.